### DISCUSSION TITLE
Shadow system behind objects

### DISCUSSION THREAD

**Discussion by BuzzLEquerre** [flareteam/flare-game#979] [Ideas]
Hello, I'm not a programming expert but I had an idea that could be done without too much difficulty (I hope).
When the player passes behind the center of an object's tile, he appears behind it, only what would be possible to do is to make the object become transparent instead. (for exemple like in Exiled Kingdom)
![Capture d'écran 2024-12-01 132356](https://github.com/user-attachments/assets/2b53b962-1d97-46a3-83b6-2ad21ba84f5b)
![Capture d'écran 2024-12-01 132412](https://github.com/user-attachments/assets/f924fa18-99ab-4fef-b5db-388ccfa97533)

I hope I have explained my idea well

**Comment by dorkster** [comment]
I had experimented with making tiles that obscured the player (or enemies) translucent. But I felt it looked ugly, since sometimes it'd reveal parts of the map that weren't intended to be seen (e.g. no floor tiles). My solution was to draw the arrow markers over entities when they were obscured, but I'm not entirely happy with that from a gameplay perspective. They can be too hard to see sometimes.

So, I may revisit making obscuring tiles translucent. I think I could make it a little less "ugly" by having the tiles fade out instead of snapping directly to a specific opacity. Maybe based on proximity to the tile's center point?

**Reply by BuzzLEquerre** [reply]
yes why not it could be a good idea, well if it is possible of course. Ideally it would be necessary to widen the radius of transparency ? to get a better view
